backend/controllers/match.js

```diff
@@ -2,7 +2,10 @@ import { insertMatch, selectAllMatches, selectMatchById, updateMatch, deleteMatc
 
 export const createMatch = async (req, res) => {
   try {
-    const match = await insertMatch(...Object.values(req.body));
+    const userId = req.user.id;
+    const { score, result, duration, total_hits, total_misses } = req.body;
+
+    const match = await insertMatch(userId, score, result, duration, total_hits, total_misses);
```

Comment:
Esses req.user.id tá sempre retornando undefined.

```suggestion
    const match = await insertMatch(...Object.values(req.body));
```